package/src/lib/octokit.ts

```diff
@@ -1,43 +1,111 @@
-  if (!GITHUB_PERSONAL_TOKEN) {
-    return false;
-  }
-  return true;
-}
+// Create Gist Logger interface
+const gistLogger = async (
+  type: "info"|"warn"|"error", 
+  message: string,
+  VerboseCheck: boolean
+  ) => {
+    // if checkVerbose is true and isVerbose is true, log the message
+    if (VerboseCheck && isVerbose) {
+      if (type === "info") {
+        console.log(`[astro-gists : octokit] ${message}`);
+      } else if (type === "warn") {
+        console.log(`[WARN] [astro-gists : octokit] ${message}`);
+      } else if (type === "error") {
+        console.log(`[ERROR] [astro-gists : octokit] ${message}`);
+      } 
+    }
 
-export const TOKEN_MISSING_ERROR = "GITHUB_PERSONAL_TOKEN not found. Please add it to your .env file. Without it, you will be limited to 60 requests per hour.";
+    if (!VerboseCheck) {
+      if (type === "info") {
+        console.log(`[astro-gists : octokit]" ${message}`);
+      } else if (type === "warn") {
+        console.log(`[WARN] [astro-gists : octokit] ${message}`);
+      } else if (type === "error") {
+        console.log(`[ERROR] [astro-gists : octokit] ${message}`);
+      }
+    }
+
+  };
+
+// Load environment variables
+const { GITHUB_PERSONAL_TOKEN } = loadEnv("all", process.cwd(), "GITHUB_");
 
 // Create an Octokit instance
 const octokit = new Octokit({ auth: GITHUB_PERSONAL_TOKEN });
 
-// Retry requests if rate limited
-export async function requestRetry(route: Route, parameters: RequestParameters) {
-  try {
-    const response: OctokitResponse<unknown, number> = await octokit.request(route, parameters);
-    return response;
-  } catch (error) {
-    /** @ts-ignore-error */
-    if (error.response && error.status === 403 && error.response.headers['x-ratelimit-remaining'] === '0') {
-      /** @ts-ignore-error */
-      const resetTimeEpochSeconds = error.response.headers['x-ratelimit-reset'];
-      const currentTimeEpochSeconds = Math.floor(new Date().getTime() / 1000);
-      const secondsToWait = resetTimeEpochSeconds - currentTimeEpochSeconds;
-      console.log(`Rate limit reached. Waiting ${secondsToWait} seconds before retrying.`);
-      return new Promise((resolve) => {
-        setTimeout(async () => {
-          const retryResponse = await requestRetry(route, parameters);
-          resolve(retryResponse);
-        }, secondsToWait * 1000);
-      });
-    }
-      // Return a rejected Promise
-      return Promise.reject(error);
+// Retry failed requests
+const retry: typeof pRretry = (fn, opts) =>
+  pRretry(fn, {
+    onFailedAttempt: (e) =>
+      gistLogger("warn",
+        `Attempt ${e.attemptNumber} failed. There are ${e.retriesLeft} retries left.\n ${e.message}`,
+        false),
+      retries: 3,
+    ...opts,
+  });
+
+// Handle the response from the Octokit API
+// biome-ignore lint/suspicious/noExplicitAny: any is used to handle the response from the Octokit API
+function handleResponse(response: OctokitResponse<any>) {
```

Comment:
Hmmmmmmm this isn't really "handling" the response, it's just mapping the status code. Could be called `getStatusCode`